### DISCUSSION TITLE
Refresh changes after restore

### DISCUSSION THREAD

**Discussion by howdu** [TappNetwork/filament-auditing#4] [Q&A] [answered]
This is amazing exactly what we needed - thank you! 

Only issue finding is that after clicking restore button you need to do a manual refresh to see the changes. 

FYI using this to display relationship changes really shocked how well it just work. 
```php
protected function getTableQuery(): Builder
{
    $query = parent::getTableQuery();

    $page_content_ids = $this->getOwnerRecord()->pageContents()->pluck('id')->toArray();
    $query->orWhere(fn($builder) => $builder->where('auditable_type', PageContent::class)->whereIn('auditable_id', $page_content_ids));

    return $query;
}
```

**Comment by andreia** [comment, marked as answer]
@howdu I'm glad you like it! :)

In this [new release](https://github.com/TappNetwork/filament-auditing/releases/tag/v1.5) the `auditRestored` event has been added (emitted when the audit is restored) so you can listen to it and use it as you see fit.

I was testing the Livewire's [`$refresh`](https://laravel-livewire.com/docs/2.x/actions#magic-actions) on the [Edit page](https://filamentphp.com/docs/2.x/admin/resources/editing-records), but isn't working using directly on listeners: 

```php
protected $listeners = ['auditRestored' => '$refresh'];
```

or in a custom method:

```php
protected $listeners = ['auditRestored'];

public function auditRestored()
{
    $refresh;
}
```

Using something like this is working:

```php
public function auditRestored()
{
    $attributes = $this->record->attributesToArray();
    $data = $this->mutateFormDataBeforeFill($attributes);

    $this->form->fill($data);
}
```

Please let me know if it works for you.

Thank you and apologies for the delay!

**Reply by howdu** [reply]
Works brilliantly thank you!
